comms/src/peer_manager/peer_storage.rs

```diff
@@ -393,7 +393,7 @@ impl PeerStorage {
             .ok_or(PeerManagerError::PeerNotFoundError)?;
         peer.addresses
             .mark_message_received(net_address)
-            .map_err(|_| PeerManagerError::DataUpdateError)?;
```

Comment:
Many of these errors are not necessarily NetAddressError

Reply:
Actually they all were - this change helped me find an issue, as the DataUpdateError is pretty vague